### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ccciudatu**
arrow 53.1.0
datafusion 42.0.0

### CODE REVIEW

**Comment by ccciudatu** [comment]
@phillipleblanc Hmm, just noticed the upgrade is also part of https://github.com/datafusion-contrib/datafusion-table-providers/pull/112.
Won't close this though, as it also switches to the upstream flight sql client (and removes the copy/pasted code). So I'll rebase once the other PR is merged (unless we find any reason to merge this one first).

**Comment by phillipleblanc** [comment]
@ccciudatu we can merge this first. The other PR is waiting on duckdb-rs to release a new version, which may take a while.

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**
